server/channels/api4/channel_bookmark.go

```diff
@@ -404,3 +405,30 @@ func listChannelBookmarksForChannel(c *Context, w http.ResponseWriter, r *http.R
 		c.Logger.Warn("Error while writing response", mlog.Err(err))
 	}
 }
+
+func fetchOpenGraph(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
Put `GetOpenGraphMetadata` behind  `ServiceSettings.EnableLinkPreviews` and `isLinkAllowedForPreview`, which is what `parseLinkMetadata` is secured by, I believe. 

cc @jupenur would you mind taking a look at this?

Reply:
It would've been good to keep it as the same implementation, and to make requests client-side to reduce a small bit of network activity on the server, but `fetchRaw` throws CORS from my testing

```
Access to fetch at 'https://visualstudio.com/' from origin 'https://cr-dev.test.mattermost.cloud' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.
```